app/jobs/submission_reversal_invoice_creation_job.rb

```diff
@@ -0,0 +1,12 @@
+class SubmissionReversalInvoiceCreationJob < ApplicationJob
+  def perform(submission)
+    raise "Submission #{submission.id} already has a reversal invoice." if submission.reversal_invoice.present?
+
+    workday_reference = if submission.management_charge.negative?
+                          Workday::SubmitCustomerReversalInvoiceRequest.new(submission).perform
+                        else
+                          Workday::SubmitCustomerReversalInvoiceAdjustmentRequest.new(submission).perform
+                        end
+    submission.create_reversal_invoice!(workday_reference: workday_reference)
```

Comment:
if saving this record fails, what happens when the job is re-run? Will Workday let us create a second reversal?